pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/TableConfigsRestletResource.java

```diff
@@ -0,0 +1,358 @@
+  /**
+   * Deletes the {@link TableConfigs} by deleting the schema tableName, the offline table config for tableName_OFFLINE and
+   * the realtime table config for tableName_REALTIME
+   */
+  @DELETE
+  @Path("/tableConfigs/{tableName}")
+  @Authenticate(AccessType.DELETE)
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Delete the TableConfigs", notes = "Delete the TableConfigs")
+  public SuccessResponse deleteConfig(
+      @ApiParam(value = "TableConfigs name i.e. raw table name", required = true) @PathParam("tableName") String tableName) {
+
+    try {
+      _pinotHelixResourceManager.deleteRealtimeTable(tableName);
+      LOGGER.info("Deleted realtime table: {}", tableName);
+      _pinotHelixResourceManager.deleteOfflineTable(tableName);
+      LOGGER.info("Deleted offline table: {}", tableName);
+      Schema schema = _pinotHelixResourceManager.getSchema(tableName);
+      if (schema != null) {
+        _pinotHelixResourceManager.deleteSchema(schema);
+      }
+      LOGGER.info("Deleted schema: {}", tableName);
+      return new SuccessResponse("Deleted TableConfigs: " + tableName);
+    } catch (Exception e) {
+      throw new ControllerApplicationException(LOGGER, e.getMessage(), Response.Status.INTERNAL_SERVER_ERROR, e);
+    }
+  }
+
+  /**
+   * Updated the {@link TableConfigs} by updating the schema tableName,
+   * then updating the offline tableConfig or creating a new one if it doesn't already exist in the cluster,
+   * then updating the realtime tableConfig or creating a new one if it doesn't already exist in the cluster.
+   */
+  @PUT
+  @Path("/tableConfigs/{tableName}")
+  @Authenticate(AccessType.UPDATE)
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Update the TableConfigs provided by the tableConfigsStr json", notes = "Update the TableConfigs provided by the tableConfigsStr json")
+  public SuccessResponse updateConfig(
+      @ApiParam(value = "TableConfigs name i.e. raw table name", required = true) @PathParam("tableName") String tableName,
+      @ApiParam(value = "Reload the table if the new schema is backward compatible") @DefaultValue("false") @QueryParam("reload") boolean reload,
+      String tableConfigsStr)
+      throws Exception {
+    TableConfigs tableConfigs;
+    try {
+      tableConfigs = JsonUtils.stringToObject(tableConfigsStr, TableConfigs.class);
+      Preconditions.checkState(tableConfigs.getTableName().equals(tableName),
+          "'tableName' in TableConfigs: %s must match provided tableName: %s", tableConfigs.getTableName(), tableName);
+      validateConfig(tableConfigs);
+    } catch (Exception e) {
+      throw new ControllerApplicationException(LOGGER, String.format("Invalid TableConfigs: %s", tableName),
+          Response.Status.BAD_REQUEST, e);
+    }
+
+    TableConfig offlineTableConfig = tableConfigs.getOffline();
+    TableConfig realtimeTableConfig = tableConfigs.getRealtime();
+    Schema schema = tableConfigs.getSchema();
+
+    try {
```

Comment:
Check table existence first

Reply:
Done, checking for existence of tables before allowing PUT